### PR TITLE
Update `@JsonFormat` javaDoc regarding `LocalDate` usage.

### PR DESCRIPTION
**Pull request by JooHyukKim**
Resolves [jackson-modules-java8#330](https://github.com/FasterXML/jackson-modules-java8/issues/330)

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/annotation/JsonFormat.java [hunk 1]
```diff
@@ -22,6 +22,11 @@
  * <li>{@link java.util.Date}: Shape can  be {@link Shape#STRING} or {@link Shape#NUMBER};
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/annotation/JsonFormat.java:22]: Not part of change here, but should probably add `java.util.Calendar` along with `java.util.Date`.

**Review by cowtowncoder — APPROVED** [review]
Looks good wrt additions; added suggestions for couple of other small changes.

**Comment by JooHyukKim** [comment]
@cowtowncoder added suggestions.
FYI, alongside `java.util.Calendar` and `java.util.Date`, there are also `java.sql.Date` and `java.sql.Timestamp` as well

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
@JooHyukKim yes. Those are not that commonly used so let's start with main types.

**Pull request merged**